cdap-ui/app/cdap/components/DataPrepHome/index.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+import NavigationPrompt from 'react-router/NavigationPrompt';
+import DataPrep from 'components/DataPrep';
+
+/**
+ *  Routing container for DataPrep for React
+ **/
+export default class DataPrepHome extends Component {
+  constructor(props) {
+    super(props);
+
+    window.onbeforeunload = function() {
```

Comment:
I'm going to remove the onbeforeunload for now.. Will need to think about a better way to control this just from a single place